ciborium/src/de/mod.rs

```diff
@@ -99,7 +106,21 @@ where
                         while let Some(chunk) = segment.pull(&mut buffer)? {
                             for b in chunk {
                                 match index {
-                                    16 => return Err(de::Error::custom("bigint too large")),
+                                    16 => {
+                                        if let Some(app) = append.as_mut() {
+                                            for v in value {
+                                                app(v);
+                                            }
+                                            app(*b);
+                                            index = 17;
+                                            continue;
+                                        }
+                                        return Err(de::Error::custom("bigint too large"));
+                                    }
+                                    17 => {
+                                        append.as_mut().unwrap()(*b);
```

Comment:
Will `append` always be `Some` when `index == 17`? I'm unsure of the `.unwrap()`

Reply:
Added comments which clarify.

Reply:
Maybe could you make it `.expect()` so that a useful error message might be displayed in case `index` is ever 17 without it having been 16, or without `Some(append)`?

Reply:
I switched from using `Option` to having two parameters, though it's always fine to call append, so the boolean is just an optimization which allows for stopping early.